### ISSUE TITLE
Docs are wrong, Visual Studio 2022 DOESNT use port 4026 for remote debugging

### ISSUE THREAD

**Issue by RDavis3000** [MicrosoftDocs/visualstudio-docs#8224]
Visual Studio 2022 DOESNT use port 4026 for remote debugging

It uses 4024 same as VS2019 as of 17.2.5

Just wasted 4 hours until a random SO prompted me to try the 'wrong' port and it works!

https://stackoverflow.com/questions/69549657/visual-studio-2022-debug-on-remote-azure-server

Microsoft Visual Studio Community 2022
Version 17.2.5
VisualStudio.17.Release/17.2.5+32616.157
Microsoft .NET Framework
Version 4.8.04161

Installed Version: Community

.NET Core Debugging with WSL   1.0
.NET Core Debugging with WSL

ADL Tools Service Provider   1.0
This package contains services used by Data Lake tools

ASA Service Provider   1.0

ASP.NET and Web Tools 2019   17.2.393.26812
ASP.NET and Web Tools 2019

Azure App Service Tools v3.0.0   17.2.393.26812
Azure App Service Tools v3.0.0

Azure Data Lake Tools for Visual Studio   2.6.5000.0
Microsoft Azure Data Lake Tools for Visual Studio

Azure Functions and Web Jobs Tools   17.2.393.26812
Azure Functions and Web Jobs Tools

Azure Stream Analytics Tools for Visual Studio   2.6.5000.0
Microsoft Azure Stream Analytics Tools for Visual Studio

C# Tools   4.2.0-4.22281.5+8d3180e5f00d42f0f0295165f756f368f0cbfa44
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Microsoft Azure Hive Query Language Service   2.6.5000.0
Language service for Hive query

Microsoft Azure Stream Analytics Language Service   2.6.5000.0
Language service for Azure Stream Analytics

Microsoft Azure Tools for Visual Studio   2.9
Support for Azure Cloud Services projects

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

NuGet Package Manager   6.2.1
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Razor (ASP.NET Core)   17.0.0.2218101+885a343b00bcab620a90c1550c37dafd730ce984
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   17.0.62204.01010
Microsoft SQL Server Data Tools

ToolWindowHostedEditor   1.0
Hosting json editor into a tool window

TypeScript Tools   17.0.10418.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.2.0-4.22281.5+8d3180e5f00d42f0f0295165f756f368f0cbfa44
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.1.0-beta.22274.4+211fd760c2315d73cfa0d375fe64abb346bc1096
Microsoft Visual F# Tools

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 75c19e4f-4623-c462-e399-008a85ea1523
* Version Independent ID: bc0b82bb-acc8-7438-7b18-5165de39d847
* Content: [Remote Debugger Port Assignments - Visual Studio (Windows)](https://docs.microsoft.com/en-us/visualstudio/debugger/remote-debugger-port-assignments?view=vs-2019)
* Content Source: [docs/debugger/remote-debugger-port-assignments.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/main/docs/debugger/remote-debugger-port-assignments.md)
* Product: **visual-studio-windows**
* Technology: **vs-ide-debug**
* GitHub Login: @Mikejo5000
* Microsoft Alias: **mikejo**

**Comment by Mikejo5000** [comment]
@gregg-miskelly to comment. I wonder if Azure is still running the older version of the Remote Debugger, so 4024 works instead of 4026.

**Comment by WilliamAntonRohm** [comment]
@gregg-miskelly -- Gregg, thank you for clarifying.

I am closing this issue now.  You are welcome to @ mention me for any followup.  We hope to hear from you again.

**Comment by RDavis3000** [comment]
> The debugger is always able to use any port -- msvsmon can be configured either through the command line or configuration UI, and in Visual Studio you can tell it what port to connect to using the `<hostname>:<port-number>` syntax.
> 
> In Visual Studio 2022, the **default** port changed to 4026. However, Azure doesn't use the default port any longer -- they use 4024+4025 for both VS 2019 and VS 2022 and most likely for all future versions.

@WilliamAntonRohm I think you have misinterpreted @gregg-miskelly answer.

TL;DR; Right now the docs imply that I should use 4026 to connect to Azure from VS2022, but it doesn't work because Azure is listening on 4024 and 402 (as per @gregg-miskelly comment)

Its not relevant that the port is configurable, the point is that the documentation instructs users to use the wrong configuration.
If I use the information from either:

https://docs.microsoft.com/en-us/visualstudio/debugger/remote-debugger-port-assignments?view=vs-2022
https://docs.microsoft.com/en-us/visualstudio/debugger/configure-the-windows-firewall-for-remote-debugging?view=vs-2022

and configure my Visual Studio debugger to port 4026, or leave it blank in the dialog (thus defaulting to 4026 as per gregg's comment) the debugger will not connect because Azure is using ports 4024+4025 (as per gregg's comment) and NOT 4026 despite the documentation saying 4026 is the correct configuration.

Does that make sense?

Please re-open and rectify, or can someone explain how documenting 4026 when reality is 4024 or 4025 is somehow ok?

**Comment by WilliamAntonRohm** [comment]
@RDavis3000 -- thank you for your follow-up.

@Mikejo5000 -- please look into this issue.

**Comment by gregg-miskelly** [comment]
For what it's worth, IMHO:

* The main section of the doc where it lists the default port number for the remote debugger service is correct
* The [Remote Debugger Ports on Azure](https://github.com/MicrosoftDocs/visualstudio-docs/blob/main/docs/debugger/remote-debugger-port-assignments.md#remote-debugger-ports-on-azure) section at the bottom is confusing. I **think** that is for the Azure VM remote debugging that I don't think exists anymore. We don't have a section for Windows Azure App Service, which is by far the most common way to do remote debugging, and, as this issue points out, is confusing.

**Comment by RDavis3000** [comment]
A far more measured response 😊

All I know is that the Microsoft Docs didn’t help me get it running but a Stack Overflow comment did. In fact if I hadn’t bothered to read the MS docs and just used 4024 as I previously did with VS2019 it would have saved me a few hours! I was measurably and materially worse off by having read the documentation, which seems less than ideal 😊

Also, this is not an uncommon use case, and I’m not an uncommonly dense dev, so help us out a little?

**Comment by Mikejo5000** [comment]
Thanks, @gregg-miskelly and @RDavis3000 for the help with this. I'll update the doc best as we can based on Gregg's comment.

**Comment by cboynton-ms** [comment]
The issue is not completely resolved with the documentation. Please see: https://learn.microsoft.com/en-us/visualstudio/debugger/remote-debugger-port-assignments?view=vs-2022

> The Visual Studio Remote Debugger can run as an application or as a background service. When it runs as an application, it uses a port that is assigned by default as follows:
> 
> Visual Studio 2022: 4026
> Visual Studio 2019: 4024
> Visual Studio 2017: 4022
> Visual Studio 2015: 4020
> Visual Studio 2013: 4018
> Visual Studio 2012: 4016

> TCP 4026 (in Visual Studio 2022) is the main port, and is required for all scenarios excluding Azure App Service.

etc. etc.
